django/publicmapping/redistricting/config.py

```diff
@@ -1288,20 +1563,24 @@ def _check_spatial_resource(self, url, name, dictionary, update=False):
         @keyword update: Optional. Update the featuretype if it exists?
         @returns: A flag indicating if the configuration call completed successfully.
         """
+        # TODO: this throws a 404 from geoserver every time it can't find something,
+        # and we expect it not to be able to find something pretty often. While I wouldn't
```

Comment:
What does this mean exactly? Are these geoserver endpoints just flakey?

Reply:
No, it's by design -- `_rest_check` attempts to read a resource that, if it existed, would have a known name and the geoserver routes use the resource names. If it's there, there's logic to avoid trying to create it. If it's not there, it creates it. It discovers that it's not there through the 404 though. It's not that it's broken, just that there's a really low signal-to-noise ratio in the geoserver container's stdout.